pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -541,10 +541,12 @@ def _binary_op_method_timedeltalike(op, name):
 
         elif hasattr(other, 'dtype'):
             # nd-array like
-            if other.dtype.kind not in ['m', 'M']:
-                # raise rathering than letting numpy return wrong answer
+            if other.dtype.kind in ['m', 'M']:
+                return op(self.to_timedelta64(), other)
+            elif other.dtype.kind == 'O':
+                return np.array([op(self, x) for x in other])
```

Comment:
@jbrockmendel  I am a bit confused here why simply returning NotImplemented is not sufficient ? 
(I tested it, and that doesn't seem to work. Although with a `datetime.timedelta` it does, and that one *does* return NotImplemented ..)

Reply:
The `pd.Timedelta` version fails because both `arr.__add__(td)` and `td.__radd__(arr)` return `NotImplemented`.  `arr.__add__(td.to_pytimedelta())` returns OK, so presumably it is something on the numpy implementation.

Reply:
Was playing a bit with it, and I think Timedelta behaves differently as datetime.timedelta because of the `__array_priority__` we add to the `_Timedelta` class (by commenting it out, the simple example works), which is needed to get other behaviors working I suppose

Reply:
@jorisvandenbossche ok otherwise this PR looks ok, are you suggesting that we remove that?

Reply:
I'm also confused.